### ISSUE TITLE
Better handling of "additional" files

### ISSUE THREAD

**Issue by rarkins** [renovatebot/renovate#5608]
We now have a few cases, such as in Gradle and Maven, where Renovate may extract and update a file which is not the original "package file".

In some cases we require users to define `fileMatch` for the extra file, while in others we find and later update it automatically.

I noticed a use case from `ansible-galaxy` that should make us consider if we make this concept have first-class support, e.g. file a new field.

Docs: https://galaxy.ansible.com/docs/using/installing.html#multiple-roles-from-multiple-files

Screenshot:
![image](https://user-images.githubusercontent.com/6311784/75622440-8d324b80-5ba0-11ea-82b2-852ea8367438.png)

Today a Renovate user would need to extend their `fileMatch` to include this `<path_to_requirements>/webserver.yml` file, but in theory the better use case would be if the `ansible-manager` could detect and parse it automatically. The question I have is how we can "de-duplicate" our results in such a case. Possibilities:
- Allow the extract function to return an array that essentially "adds" new package files, OR
- Return all found deps mapped to the original `requirements.yaml`, but deps can optionally include a new field named something like `foundInFile` that would e.g. be `<path_to_requirements>/webserver.yml` in this case

**Comment by viceice** [comment]
I think adding the file property, where the change was found is the better idea. With that we can use our generic replacement to update the file and add it to the commit.

**Comment by rarkins** [comment]
Yes that was part of the idea. Would make generic replacement much easier!

**Comment by secustor** [comment]
Are the `propSource` and `editFile` properties in anyway connected to this? I could only find that these are used internally of the maven manager.

**Comment by rarkins** [comment]
@secustor `editFile` was intended to be connected to this, although currently our auto replace functionality doesn't work yet with it, so things are a little stuck.

The ultimate idea of this issue is that:
- Version strings may be in "other" files and those other files need updating
- We shouldn't need to call those other files "package files" because that's not what they are
